content/en/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md

```diff
@@ -0,0 +1,115 @@
+
+## Preventing unauthorised users from converting the volume mode
+
+In this context, an authorised user is one who has access rights to perform **update**
+or **patch** operations on VolumeSnapshotContents, which is a cluster-level resource.  
+It is up to the cluster administrator to provide these rights only to trusted users
+or applications, like backup vendors.
+Users apart from such authorised ones will never be allowed to modify the volume mode
+of a PVC when it is being created from a VolumeSnapshot.
+
+To convert the volume mode, an authorised user must do the following:
+
+1. Identify the VolumeSnapshot that is to be used as the data source for a newly
+   created PVC in the given namespace.
+2. Identify the VolumeSnapshotContent bound to the above VolumeSnapshot.
+
+  ```shell
+  kubectl describe volumesnapshot -n <namespace> <name>
+  ```
+
+3. Add the annotation [`snapshot.storage.kubernetes.io/allow-volume-mode-change: "true"`](/docs/reference/labels-annotations-taints/#snapshot-storage-kubernetes-io-allowvolumemodechange)
+   to the above VolumeSnapshotContent. The VolumeSnapshotContent annotations must include one similar to the following manifest fragment:
+
+  ```yaml
+  kind: VolumeSnapshotContent
+  metadata:
+    annotations:
+      - snapshot.storage.kubernetes.io/allow-volume-mode-change: "true"
+  ...
+  ```
+
+**Note**: For pre-provisioned VolumeSnapshotContents, you must take an extra
+step of setting `spec.sourceVolumeMode` field to either `Filesystem` or `Block`,
+depending on the mode of the volume from which this snapshot was taken.
+
+An example is shown below:
+
+   ```yaml
+   apiVersion: snapshot.storage.k8s.io/v1
+   kind: VolumeSnapshotContent
+   metadata:
+     annotations:
+     - snapshot.storage.kubernetes.io/allow-volume-mode-change: "true"
+     name: <volume-snapshot-content-name>
+   spec:
+     deletionPolicy: Delete
+     driver: hostpath.csi.k8s.io
+     source:
+       snapshotHandle: <snapshot-handle>
+     sourceVolumeMode: Filesystem
+     volumeSnapshotRef:
+       name: <volume-snapshot-name>
+       namespace: <namespace>
+   ```
+
+Repeat steps 1 to 3 for all VolumeSnapshotContents whose volume mode needs to be
+converted during a backup or restore operation. This can be done either via software
+with credentials of an authorised user or manually by the authorised user(s).
+
+If the annotation shown above is present on a VolumeSnapshotContent object,
+Kubernetes will not prevent the volume mode from being converted.
+Users should keep this in mind before they attempt to add the annotation
+to any VolumeSnapshotContent.
+
+## Action Required
+
+The `prevent-volume-mode-conversion` feature flag is enabled by default in the 
+external-provisioner `v4.0.0` and snapshot-controller `v7.0.0` CSI sidecar 
+containers. Volume mode change will be rejected when creating a PVC from
+a VolumeSnapshot unless the steps described above have been performed.
```

Comment:
One more comment: can you just change to "external-provisioner `v4.0.0` and external-snapshotter `v7.0.0`?  The flag is available in snapshot-controller but not really available in csi-snapshotter sidecar.